client-js/app/scripts/registry.ts

```diff
@@ -9,182 +11,10 @@ export interface RegistryListing {
   vals: Array<string>;
 }
 
-export interface RegistryApi {
```

Comment:
OK, thanks for explaining that. What I don't understand is if the interface is defined somewhere, why do the methods have to be defined again within the class again, it looks to me like the same information is being written twice.